python/lsst/daf/butler/core/datastore.py

```diff
@@ -63,19 +65,17 @@ class DatastoreTransaction:
     ----------
     parent : `DatastoreTransaction`, optional
         The parent transaction (if any)
-
-    Attributes
-    ----------
-    parent : `DatastoreTransaction`
-        The parent transaction.
     """
-    Event = Event
+    Event: ClassVar[Type] = Event
+
+    parent: Optional['DatastoreTransaction']
```

Comment:
At the risk of sending you down another rabbit hole, if you do `from __future__ import annotations`, you don't need to use a string here.  As a heads-up, if you do that, you'll need to bump the Travis Python version to 3.7.